src/Azure.Functions.Cli/Actions/DeployActions/Platforms/KnativePlatform.cs

```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Linq;
+using System.IO;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Functions.Cli.Common;
+using Azure.Functions.Cli.Helpers;
+using Azure.Functions.Cli.Interfaces;
+using Colors.Net;
+using KubeClient;
+using KubeClient.Models;
+using Azure.Functions.Cli.Actions.DeployActions.Platforms.Models;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+
+namespace Azure.Functions.Cli.Actions.DeployActions.Platforms
+{
+    public class KnativePlatform : IHostingPlatform
+    {
+        private string configFile = string.Empty;
+        private const string FUNCTIONS_NAMESPACE = "azure-functions";
+        private static KubeApiClient client;
+
+        public async Task DeployContainerizedFunction(string functionName, string image, int min, int max)
+        {
+            await Deploy(functionName, image, FUNCTIONS_NAMESPACE, min, max);
+        }
+
+        public KnativePlatform(string configFile)
+        {
+            this.configFile = configFile;
+            KubeClientOptions options;
+
+            if (!string.IsNullOrEmpty(configFile))
+            {
+                options = K8sConfig.Load(configFile).ToKubeClientOptions(defaultKubeNamespace: FUNCTIONS_NAMESPACE);
+            }
+            else
+            {
+                options = K8sConfig.Load().ToKubeClientOptions(defaultKubeNamespace: FUNCTIONS_NAMESPACE);
+            }
+
+            client = KubeApiClient.Create(options);
+        }
+
+        private async Task Deploy(string name, string image, string nameSpace, int min, int max)
```

Comment:
I'm a bit unclear about the use of this `name` here vs in the `KubernetesPlatform.cs`. In `KubernetesPlatform` it seems that name is used to name the deployment, while here, that name is used in the `IsHttpTrigger` method to check if that one function is in fact http, _as well as_ used as a name for the deployment. 

Shouldn't the check for `IsHttpTrigger` really be `IsAnyHttpTrigger` and then there enumerate all `*/function.json`

Reply:
The deploy command deploys a single function, thus a single function.json